lib/tests/streamlit/web/cli_test.py

```diff
@@ -285,6 +285,28 @@ def test_credentials_headless_with_config(self, headless_mode):
         self.assertTrue(mock_check.called)
         self.assertEqual(0, result.exit_code)
 
+    @parameterized.expand([(True,), (False,)])
+    def test_headless_telemetry_message(self, headless_mode):
+        """If headless mode, show a message about usage metrics gathering."""
+
+        from streamlit import config
+
+        config._set_option("server.headless", headless_mode, "test")
```

Comment:
Nit: this should use `patch_config_options` (which resets the modified options when complete):

```
with patch_config_options({"server.headless": headless_mode}):
   do the stuff
```

Reply:
Nice, I did not know about that :) I also updated the two other tests with `_set_option` in cli_test.